src/mlagility/cli/report.py

```diff
@@ -1,521 +1,240 @@
-    printing.log("Build results across all models collected at ")
-    printing.logn(str(results_file), printing.Colors.OKGREEN)
+    # Description: An estimation of latency, in ms, to run one invocation of the
+    #              model on a GroqNode server (includes I/O).
+    # Source: computed by GroqFlow
+    groq_estimated_latency: str = "-"
 
+    # Description: Mean total latency, in ms, to run one invocation using TensorRT.
+    # Source: benchit.benchmark() on 100 runs
+    nvidia_latency: str = "-"
 
-def op_spreadsheet(cache_dir):
+    # Description: Mean total latency, in ms, to run one invocation using OnnxRuntime.
+    # Source: benchit.benchmark() on 100 runs
+    x86_latency: str = "-"
 
-    # Declare file paths
-    spreadsheet_file = os.path.join(cache_dir, op_support_filename)
 
-    with open(spreadsheet_file, "w", newline="", encoding="utf8") as spreadsheet:
-        writer = csv.writer(spreadsheet)
+def summary_spreadsheet(args) -> None:
+
+    # Input arguments from CLI
+    cache_dirs = [os.path.expanduser(dir) for dir in args.cache_dirs]
+    report_dir = os.path.expanduser(args.report_dir)
 
-        # Git commit placeholder
-        writer.writerow(
-            [
-                "Git Commit: ",
-            ]
-        )
+    # Name report file
+    report_path = os.path.join(report_dir, get_report_name())
 
-        writer.writerow(
-            [
-                "Model Name",
-                "Unsupported ops (Optimized ONNX)",
-            ]
-        )
+    # Create report dict
+    Path(report_dir).mkdir(parents=True, exist_ok=True)
+    report = {}
 
+    # Add results from all user-provided cache folders
+    for cache_dir in cache_dirs:
+
+        # Check if this is a valid cache directory
+        check_cache_dir(cache_dir)
+
+        # List all yaml files available
         all_model_state_yamls = cache.get_all(path=cache_dir, file_type="state.yaml")
         all_model_state_yamls = sorted(all_model_state_yamls)
 
+        # Add each model to report
         for model_state_yaml in all_model_state_yamls:
+
+            # Load state
             state = build.load_state(state_path=model_state_yaml)
 
-            writer.writerow(
-                [
-                    state.config.build_name,  # Model Name
-                    _cleanOpList(
-                        state.info.opt_onnx_unsupported_ops
-                    ),  # Unsupported ops (Optimized ONNX)
-                ]
+            # Models are identified my the build name
+            build_name = state.config.build_name
+
+            # Add model to report if it doesn't exist
+            if build_name not in report:
+                report[build_name] = BuildResults()
+
+            # Get model hash from build name
+            # TODO: Get hash_model from state once it is consistently reported
+            # https://github.com/groq/mlagility/issues/149
+            report[build_name].hash = build_name.split("_")[-1]
```

Comment:
```suggestion
            report[build_name].hash = state.info.mlagility.hash
```

Reply:
Postponing this until `state.info.mlagility` actually works